### PR TITLE
#159987727 Add Favourite Article functionality

### PR DESCRIPTION
**Pull request by d-beloved**
#### What does this PR do?
- Add ability for a user to be able to favorite certain articles
- List the articles favorited by a user
- deletes articles favorited by a user

#### How should this be manually tested?
Run 'npm install'
create a .env using the .env.sample template
Run 'npm run migrate'
Run 'npm run start:dev'
use postman to test:
```routes added```
post: /api/articles/:articlesId/favorite
delete: /api/articles/:articlesId/favorite
get: /api/user/articles/favorite

#### What are the relevant pivotal tracker stories?
 #159987727

### CODE REVIEW

**Review by gonuoha — CHANGES_REQUESTED** [review]
please change you error responses to conform to the format we have been using ... {
status: 'error',
errors: {message: '...'}
}

**Review by emekafredy — CHANGES_REQUESTED** [review]
Please make sure to update the swagger docs file with your endpoints. reach out to @lauragift21 on this if you need help.

File: server/routes/api/favoriteArticle.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Router } from 'express';
+import favoriteArticleController from '../../controllers/favoriteArticle';
+import auth from '../../middleware/auth';
```
Comment (aknwosu) [inline server/routes/api/favoriteArticle.js:3]: import {authenticateUser} from '../../middleware/auth';

File: server/routes/api/favoriteArticle.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const router = Router();
+
+// methods that authenticates the user
+const { authenticateUser } = auth;
```
Comment (aknwosu) [inline server/routes/api/favoriteArticle.js]: remove this

Comment (aknwosu) [inline server/routes/api/favoriteArticle.js]: you can remove this after modifying line 2

**Review by aknwosu — CHANGES_REQUESTED** [review]

File: server/routes/api/favoriteArticle.js (hunk 1 above)
Comment (d-beloved) [reply server/routes/api/favoriteArticle.js:3]: Thanks, I've effected this change

**Review by emekafredy — APPROVED** [review]
LGTM

**Review by gonuoha — APPROVED** [review]

**Review by aknwosu — APPROVED** [review]

**Pull request merged**